sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collationExpressions.scala

```diff
@@ -56,7 +57,8 @@ object CollateExpressionBuilder extends ExpressionBuilder {
                 evalCollation.toString.toUpperCase().contains("TRIM")) {
                 throw QueryCompilationErrors.trimCollationNotEnabledError()
               }
-              Collate(e, evalCollation.toString)
+              Collate(e, UnresolvedCollation(
```

Comment:
Why do we have `CollateExpressionBuilder`? The COLLATE function has a dedicate SQL syntax and doesn't need a function registration.

Reply:
> The COLLATE function has a dedicate SQL syntax and doesn't need a function registration.

I Agree. I propose we fix this in a followup PR.